Mathlib/LinearAlgebra/CliffordAlgebra/Prod.lean

```diff
@@ -0,0 +1,141 @@
+  | hp i₁' m₁' hm₁ =>
+    obtain ⟨i₁n, rfl⟩ := i₁'
+    dsimp only at *
+    induction hm₁ using Submodule.pow_induction_on_left' with
+    | hr =>
+      rw [AlgHom.commutes, Nat.cast_zero, mul_zero, uzpow_zero, one_smul, Algebra.commutes]
+    | hadd _ _ _ _ _ ihx ihy =>
+      rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+    | hmul m₁ hm₁ i x₁ _hx₁ ih₁ =>
+      obtain ⟨v₁, rfl⟩ := hm₁
+      -- this is the first interesting goal
+      rw [map_mul, mul_assoc, ih₁, mul_smul_comm, map_apply_ι, Nat.cast_succ, mul_add_one,
+        uzpow_add, mul_smul, ← mul_assoc, ← mul_assoc, ← smul_mul_assoc ((-1) ^ i₂),
+        QuadraticForm.Isometry.inl_apply]
+      clear ih₁
+      congr 2
+      induction hm₂ using Submodule.iSup_induction' with
+      | h0 => rw [map_zero, zero_mul, mul_zero, smul_zero]
+      | hadd _ _ _ _ ihx ihy => rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+      | hp i₂' m₂' hm₂ =>
+        clear m₂
+        obtain ⟨i₂n, rfl⟩ := i₂'
+        dsimp only at *
+        induction hm₂ using Submodule.pow_induction_on_left' with
+        | hr =>
+          rw [AlgHom.commutes, Nat.cast_zero, uzpow_zero, one_smul, Algebra.commutes]
+        | hadd _ _ _ _ _ ihx ihy =>
+          rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+        | hmul m₂ hm₂ i x₂ _hx₂ ih₂ =>
+          obtain ⟨v₂, rfl⟩ := hm₂
+          -- this is the second interesting goal
+          rw [map_mul, map_apply_ι, QuadraticForm.Isometry.inr_apply, Nat.cast_succ, ← mul_assoc,
+            ι_mul_ι_comm_of_isOrtho (.inl_inr _ _), neg_mul, mul_assoc, ih₂, mul_smul_comm,
+            ← mul_assoc, ← Units.neg_smul, uzpow_add, uzpow_one, mul_neg_one]
+
+
+/-- The forward direction of `CliffordAlgebra.prodEquiv`. -/
+def ofProd : CliffordAlgebra (Q₁.prod Q₂) →ₐ[R] (evenOdd Q₁ ᵍ⊗[R] evenOdd Q₂) :=
+  lift _ ⟨
+    LinearMap.coprod
+      ((GradedTensorProduct.includeLeft (evenOdd Q₁) (evenOdd Q₂)).toLinearMap
+          ∘ₗ (evenOdd Q₁ 1).subtype ∘ₗ (ι Q₁).codRestrict _ (ι_mem_evenOdd_one Q₁))
+      ((GradedTensorProduct.includeRight (evenOdd Q₁) (evenOdd Q₂)).toLinearMap
+          ∘ₗ (evenOdd Q₂ 1).subtype ∘ₗ (ι Q₂).codRestrict _ (ι_mem_evenOdd_one Q₂)),
+    fun m => by
+      dsimp only [LinearMap.coprod_apply, LinearMap.coe_comp, Function.comp_apply,
+        AlgHom.toLinearMap_apply, QuadraticForm.prod_apply, Submodule.coeSubtype,
+        GradedTensorProduct.includeLeft_apply, GradedTensorProduct.includeRight_apply]
+      simp only [map_add, add_mul, mul_add, GradedTensorProduct.algebraMap_def]
+      rw [GradedTensorProduct.tmul_one_mul_one_tmul, GradedTensorProduct.tmul_one_mul_coe_tmul,
+        GradedTensorProduct.tmul_coe_mul_one_tmul, GradedTensorProduct.tmul_coe_mul_coe_tmul]
+      dsimp
+      simp_rw [one_mul, uzpow_one, Units.neg_smul, one_smul]
+      rw [ι_sq_scalar, ι_sq_scalar, mul_one]
+      simp_rw [← GradedTensorProduct.algebraMap_def, ← GradedTensorProduct.algebraMap_def']
```

Comment:
How about:
```suggestion
      simp_rw [LinearMap.coprod_apply, LinearMap.coe_comp, Function.comp_apply,
        AlgHom.toLinearMap_apply, QuadraticForm.prod_apply, Submodule.coeSubtype,
        GradedTensorProduct.includeLeft_apply, GradedTensorProduct.includeRight_apply, map_add,
        add_mul, mul_add, GradedTensorProduct.algebraMap_def,
        GradedTensorProduct.tmul_one_mul_one_tmul, GradedTensorProduct.tmul_one_mul_coe_tmul,
        GradedTensorProduct.tmul_coe_mul_one_tmul, GradedTensorProduct.tmul_coe_mul_coe_tmul,
        LinearMap.codRestrict_apply, one_mul, uzpow_one, Units.neg_smul, one_smul, ι_sq_scalar,
        mul_one, ← GradedTensorProduct.algebraMap_def, ← GradedTensorProduct.algebraMap_def']
```
(presumably previously not possible because of now extinct shortcomings of `simp`)

Reply:
Sure, why not! I think I was trying to structure the proof a bit more, but upon re-reading I can't see what I was going for anyway.